### DISCUSSION TITLE
how to remove all on Multi-select

### DISCUSSION THREAD

**Discussion by ericthanhtung** [filamentphp/filament#11265] [Help] [answered]
### Package

Panel builder

### Package Version

v2

### How can we help you?

![image](https://github.com/filamentphp/filament/assets/95896012/011000d8-090f-4987-8426-48e64b733aff)
![image](https://github.com/filamentphp/filament/assets/95896012/8f34324a-fc09-4943-9fa3-173be99c0245)

how to remove all on Multi-select

**Comment by leandrocfe** [comment, marked as answer]
Are you using Filament v2?
Maybe a suffixAction? https://filamentphp.com/docs/2.x/forms/fields#affixes

```php
Select::make('codes')
    ->options(...)
    ->multiple()
    ->suffixAction(function (Select $component) {
        return Action::make('remove_all')
            ->icon('heroicon-s-x-circle')
            ->tooltip('Clear')
            ->action(function () use ($component) {
                $component->state([]);
            });
    })
```
